clients/admin-ui/src/features/dataset/DatabaseConnectForm.tsx

```diff
@@ -245,7 +245,11 @@ const DatabaseConnectForm = () => {
                 colorScheme="primary"
                 type="submit"
                 isLoading={isSubmitting || isLoading}
-                isDisabled={isSubmitting || isLoading}
+                isDisabled={
+                  isSubmitting ||
+                  isLoading ||
+                  features.flags.dataDiscoveryAndDetection
+                }
```

Comment:
This looks technically correct, but I think it'd be less confusing to go further up in the workflow and completely hide the "Connect to a database" option when this flag is enabled.

i.e this screen:
<img width="1178" alt="image" src="https://github.com/ethyca/fides/assets/1834295/21cbec45-66da-4976-bf86-75dd649a4131">

Reply:
adjusted this in https://github.com/ethyca/fides/pull/4972/commits/d74c03f9bb597d4d2c0fca29981f57340b00a730